File: 11t.js
function removeEgg(foods) {
    const reversed = foods.reverse(); // Reverse the array
    const result = [];
    let eggCount = 0;
  
    for (let i = 0; i < reversed.length; i++) {
      if (reversed[i] === 'egg' && eggCount < 2) {
        eggCount++; // Increment counter and skip this 'egg'
        continue;
      }
      result.push(reversed[i]); // Add non-'egg' items or excess 'egg' items
    }
  
    return result.reverse(); // Reverse the array back to the original order
  }
  
  // Example:
  console.log(removeEgg(['egg', 'apple', 'egg', 'egg', 'ham'])); // Output: ['egg', 'apple', 'ham']
  

File: 11t.py
def remove_eggs (foods):
    reversed_foods = list(reversed (foods)) # correctly reversed the foods in the array
    result = []
    egg_count = 0

    # Loop through the reversed array containing foods
    for item in reversed_foods:
        if item == 'egg' and egg_count <2:
            egg_count =+1
            continue
        result.append (item)

    # return the result reversed back to the original order
    return list (reversed (result))


print (remove_eggs(['egg', 'apple', 'egg', 'egg', 'ham'])) # print the output to the terminal